src/figuresClasses.ts

```diff
@@ -1,11 +1,78 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea: () => number;
+}
+
+function area(areaRound: number): number {
+  return Math.floor(areaRound * 100) / 100;
+}
+
+export class Triangle implements Figure {
+  public shape: Figure['shape'] = 'triangle';
+
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Enter correct values');
+    }
+
+    if (a + b <= c || b + c <= a || a + c <= b) {
+      throw new Error('Enter correct values');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const areaRound = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return area(areaRound);
+  }
+}
+
+export class Circle implements Figure {
+  public shape: Figure['shape'] = 'circle';
+
+  constructor(
+    public color: Figure['color'],
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Enter correct values');
```

Comment:
Consider providing a more detailed error message. For example: 'Circle radius must be greater than 0.'